client/src/common/components/Dropdown.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { PureComponent } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import { KeyCodes } from 'common/constants/enums';
+
+class Dropdown extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isVisible: false
+    };
+
+    this.dropdown = React.createRef();
+  }
+
+  componentDidUpdate() {
+    const { isVisible } = this.state;
+
+    if (isVisible) {
+      document.addEventListener('keydown', this.handleEscapePress);
+      document.addEventListener('click', this.handleClickOutside);
+    } else {
+      document.removeEventListener('keydown', this.handleEscapePress);
+      document.removeEventListener('click', this.handleClickOutside);
+    }
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleEscapePress);
+    document.removeEventListener('click', this.handleClickOutside);
+  }
+
+  handleEscapePress = event => {
+    const { code } = event;
+
+    if (code === KeyCodes.ESCAPE) {
+      this.setState({
+        isVisible: false
+      });
+    }
+  };
+
+  handleClickOutside = event => {
+    const isClickedOutside = !this.dropdown.current.contains(event.target);
+
+    if (isClickedOutside) {
+      this.setState({ isVisible: false });
```

Comment:
Please be consist about the new lines - here it is one liner and in line 39 you've got 3 lines.
Please use one line until prettier says otherwise.